src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -0,0 +1,64 @@
+package wooteco.subway.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import wooteco.subway.dao.LineDao;
+import wooteco.subway.domain.Line;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class LineService {
+
+    private final LineDao lineDao;
+
+    @Autowired
+    public LineService(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
```

Comment:
```suggestion
    public LineService(LineDao lineDao) {
        this.lineDao = lineDao;
    }
```
스프링의 DI 방법이 세 가지 있는데, 생성자 주입을 선호하는편입니다.
멤버 객체의 불변유지와 실행시점의 순환참조를 캐치할 수 있어요

Reply:
지금의 코드도 생성자 주입이 아닌가요?! `@Autowired` 는 개인적으로 명시성을 위해서 일부러 달아두는 편입니다!

Reply:
앗 제가 잘못생각하고 있었군요!
스프링 4.3부터 생성자주입에 Autowired를 제외해도 된다고하네요. (과거에는 사용하고 있는지는 몰랐네요 😮)

- https://stackoverflow.com/questions/41092751/spring-injects-dependencies-in-constructor-without-autowired-annotation
- https://www.baeldung.com/constructor-injection-in-spring

생성자에 Autowired를 적어야하는 번거로움과 프레임워크에 의존하는 코드를 줄일 수 있는 장점이 있는 것 같은데요.
명시성은 어떤 걸 의미하는지 알 수 있을까요?